src/rdf4cpp/util/CharMatcher.cpp

```diff
@@ -57,50 +57,67 @@ namespace rdf4cpp::util::char_matcher_detail::HWY_NAMESPACE {
         // => comparisons need to evaluate as true for the logic to work
         V const zero = Set(d, static_cast<int8_t>(ranges[0].first));
 
+#if !(HWY_TARGET & (HWY_ALL_NEON | HWY_ALL_SVE)) && !HWY_HAVE_SCALABLE
         // highway doc: on x86 < and > are 1 instruction for signed ints (3 for unsigned)
         // and <= and >= are 2 instructions regardless of signed/unsigned
+        // Set is 2 instructions, while potential load/store is 1
+
+        using single_storage = V;
+#define GET_SINGLE(x) x
+#define LOAD_SINGLE(x) Set(d, x)
+#else
+        // highway doc: on arm neon <, >, <= and >= are 1 instruction for any int
+        // Set is 1 instruction, while potential load/store is 1
+
+        using single_storage = int8_t;
+#define GET_SINGLE(x) Set(d, x)
+#define LOAD_SINGLE(x) x
+#endif
```

Comment:
This is a little confusing to me, could you explain this?

Reply:
TLDR: its an optimization i did on x86 that does not work on arm

Reply:
Ok, I reread the comments, this time they made sense to me